heidrun/smithsonian.rb

```diff
@@ -0,0 +1,446 @@
+          :as => :thumbnail_uri do
+    uri thumbnail_uri
+  end
+
+  hasView :class => DPLA::MAP::WebResource do
+    rights record.field('indexedStructured', 'online_media_rights')
+  end
+
+  # edm:isShownAt
+  #   http://collections.si.edu/search/results.htm? \
+  #     q=record_ID%3A[<record_ID>[[value]]</record_ID>]&repo=DPLA
+  isShownAt :class => DPLA::MAP::WebResource do
+    uri record.field('descriptiveNonRepeating', 'record_ID')
+              .map(&is_shown_at_uri)
+  end
+
+  # dpla:originalRecord
+  #   DPLA
+  originalRecord :class => DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  # dpla:SourceResource
+  sourceResource :class => DPLA::MAP::SourceResource do
+    # dcterms:isPartOf
+    #   <freetext category="setName" label="[n]">
+    collection :class => DPLA::MAP::Collection,
+               :each => record.field('freetext', 'setName'),
+               :as => :collection do
+      title collection
+    end
+
+    # dcterms:contributor
+    #  <name label="associated person">
+    contributor :class => DPLA::MAP::Agent,
+                :each => record.field('freetext', 'name')
+                               .match_attribute(:label, 'associated person'),
+                :as => :contributor do
+      providedLabel contributor
+    end
+
+    # dcterms:creator
+    #   <freetext category="name" label="[value]">
+    creator :class => DPLA::MAP::Agent,
+            :each => record.field('freetext', 'name')
+                           .match_attribute(:label) { |label| creator?(label) },
+            :as => :creator do
+      providedLabel creator
+    end
+
+    # dc:date
+    #   <freetext category="date" label="[value]">
+    #   *Take earliest date
+    # Gretchen says (re only the earliest date):
+    #   So the mapping need to map all dates to both `sourceResource.date`
+    #   AND `sourceResource.temporal` and then we'll run enrichment on date
+    #   in the next step.
+    date :class => DPLA::MAP::TimeSpan,
+         :each => record.field('freetext', 'date'),
+         :as => :date do
+      providedLabel date
+    end
+
+    temporal :class => DPLA::MAP::TimeSpan,
+         :each => record.field('freetext', 'date'),
+         :as => :temporal do
+      providedLabel temporal
+    end
+
+    # dcterms:description
+    #   <freetext category="notes" label="[n]">[value]
+    #   *Each instance of "notes" should be a separate value.
+    description record.field('freetext', 'notes')
+
+    # dcterms:extent
+    #   <freetext category="physicalDescription" label="Dimensions">
+    extent record.field('freetext', 'physicalDescription')
+                 .match_attribute(:label, 'Dimensions')
+
+    # dc:format
+    #   <freetext category="physicalDescription" label="Physical description">
+    #   <freetext category="physicalDescription" label="Medium">;
+    #   <object_type>
+    #
+    # JB - actually I'm seeing this in indexedStructured
```

Comment:
Sorry, that's not very clear. Just noting that <object_type> is under <indexedStructured>